src/Acme.UI/Infrastructure/Handlers/MemberSaveHandler.cs

```diff
@@ -0,0 +1,147 @@
+                {
+                    if (member.IsApproved && MandatoryFieldsArePopulated(member))
+                    {
+                        // create tree structure
+                        var parentNodeIdAndKey = GetParentNodeIdAndKey();
+                        if (!string.IsNullOrWhiteSpace(parentNodeIdAndKey.Item1) && !nodeExists(parentNodeIdAndKey.Item1, member.Name))
+                        {
+                            var node = _contentService.Create(member.Name, Guid.Parse(parentNodeIdAndKey.Item2), "physiotherapist");
+                            node.SetValue("physio_name", member.Name);
+                            node.SetValue("physio_email", member.Email);
+
+                            var address = GetUmbracoPropertyValue(member.Properties, addressAlias);
+                            var phone = GetUmbracoPropertyValue(member.Properties, phoneAlias);
+                            node.SetValue("physio_address", address);
+                            node.SetValue("physio_contact", phone);
+
+                            _contentService.SaveAndPublish(node);
+                            processedNodesIds.Add(member.Id.ToString());
+
+                            member.SetValue(memberDataFolderAlias, node.GetUdi());
+                        }
+                    }
+                }
+            }
+        }
+ 
+        private bool MandatoryFieldsArePopulated(IMember member)
+        {
+            if (string.IsNullOrWhiteSpace(member.Name) || string.IsNullOrWhiteSpace(member.Email))
+            {
+                return false;
+            }
+
+            var address = GetUmbracoPropertyValue(member.Properties, addressAlias);
+            var phone = GetUmbracoPropertyValue(member.Properties, phoneAlias);
+
+            if (string.IsNullOrWhiteSpace(address) || string.IsNullOrWhiteSpace(phone))
+            {
+                return false;
+            }
+
+            return true;
+        } 
+
+        private Tuple<string, string> GetParentNodeIdAndKey()
+        {
+            if (ExamineManager.Instance.TryGetIndex("ExternalIndex", out var index))
+            {
+                var searcher = index.GetSearcher();
+                var results = searcher.CreateQuery("content").NodeTypeAlias("physiotherapists").Execute();
+
+                if (results.Any())
+                {
+                    var node = results.First();
+                    var keyValue = node.Values.FirstOrDefault(x => "__Key".Equals(x.Key));
+                    var key = keyValue.Value;
+                    return new Tuple<string, string>(node.Id, key);
+                }
+            }
+
+            return new Tuple<string, string>(null, null);
+        }
+
+        private bool nodeExists(string parentNodeId, string nodeName)
+        {
+            if (ExamineManager.Instance.TryGetIndex("ExternalIndex", out var index))
+            {
+                var searcher = index.GetSearcher();
```

Comment:
I think we are doing the GetSearcher and CreateQuery often enough to maybe pull out a helper class for Queries?

Reply:
please have  a look at this one

Reply:
Try make this a thing that is DI'ed into the controller